sphinx_gallery/gen_rst.py

```diff
@@ -293,32 +301,120 @@ def save_figures(image_path, fig_count, gallery_conf):
             default_attr = matplotlib.rcParams['figure.' + attr]
             if to_rgba(fig_attr) != to_rgba(default_attr):
                 kwargs[attr] = fig_attr
-
-        current_fig = image_path.format(fig_count + fig_num)
-        fig.savefig(current_fig, **kwargs)
-        figure_list.append(current_fig)
+        image_paths.append(image_path)
+        fig.savefig(image_paths[-1], **kwargs)
```

Comment:
I somehow prefer to save the image first then append the filepath to the list. Do you have any reasons for this order? In my mind, if saving fails, then we don't have the image listed. In case of failure of course I expect an exception, and everything is irrelevant. I also see later on you have a check to scan for the registered images.

Reply:
I don't think the order matters. But since you prefer the other I can switch it.